src/Workspaces/Core/Portable/Workspace/Solution/Project.cs

```diff
@@ -256,7 +256,11 @@ public bool ContainsAnalyzerConfigDocument(DocumentId documentId)
 
             return ImmutableHashMapExtensions.GetOrAdd(ref _idToAnalyzerConfigDocumentMap, documentId, s_createAnalyzerConfigDocumentFunction, this);
         }
-        public async ValueTask<IEnumerable<SourceGeneratedDocument>> GetSourceGeneratedDocumentsAsync(CancellationToken cancellationToken = default)
+
+        /// <summary>
+        /// Gets all source generated documents in this project.
+        /// </summary>
+        public async ValueTask<ImmutableArray<SourceGeneratedDocument>> GetSourceGeneratedDocumentsAsync(CancellationToken cancellationToken = default)
```

Comment:
Do we have an interface to return that is immutable but not directly ImmutableArray? I worry if we end up with projects with thousands of generated files whether the larger arrays will cause issues.

Reply:
(and yes the implementation would need some tweaking in that case, but at least we're not locked in with the current signature)

Reply:
You mean we would not want to instantiate all of the SourceGeneratedDocuments?